doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+
+    - Devices Available
+    - Devices more used (more average iops) (should be an alert/trigger)
+    - Devices biggers than n Gb
+
+The inventory should also provide a way to do directly operations over physical devices:
+
+    * Identify:  Start/stop to blink the identification light
+    * Create OSD: Create an OSD using this disk device if it is available.
+    * Remove OSD: Delete the OSD using this disk device.
+
+2. Add OSDs
+===========
+
+A. Current workflow
+--------------------
+
+**CLI**:
+We can specify specific devices or use a “drive group” specification to create OSD’s in different hosts. By default, the definition of the OSDs to create is “declarative“ unless you use the unmanaged parameter.
+
+.. prompt:: bash #
+
+  ceph orch daemon add osd <host>:device1,device2 [--unmanaged=true]
+  ceph orch apply osd -i <json_file/yaml_file> [--dry-run] [--unmanaged=true]*
+
+**GUI**:
+Implemented in the dashboard section “cluster.OSDs”.
+There is a button to create the OSDs, that presents a page dialog box to select the physical devices that are going to be part of the OSDs as main, WAL or Db devices.
+It Is very difficult to make a selection ( or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
+The problem here is the UI has been designed to work with “drive groups” and not to work for the user. The “drive group” is an abstract concept that must be used only in the background. Users must not be aware of this concept.
+
+B. Proposed workflow
+--------------------
+
+**CLI and GUI**
+
+The utilization of “declarative” drive groups makes it very difficult to understand how to configure OSD’s and the implications. Also make difficult the implementation because the multiple possibilities and the big amount of different conditions that we can find in a production system makes very complex the right evaluation and use of a declarative description of the storage devices used.
+This results in unexpected situations. For example:
+* A disk cleaned can be reused automatically and without any warning for creating a new OSD.
+* New installed disks are used automatically for OSDs (without any warning)
+* Errors trying to recreate OSD’s in disk removed from the system.
+
+So there is an important thing to consider in order to simplify everything for the user and for the implementation:
+**Avoid the “declarative” use of the drive groups**
+
+**GUI**:
+
+The user should be able to define the set of physical disk devices that are going to be used to support OSD’s.
+This means to make simple things like create one OSD in a certain device, and also to define in an easy way how to create multiple OSD’s across multiple devices in different hosts.
```

Comment:
Try to do this using the dashboard

Reply:
Well, once the underlying support for this is available in `cephadm`, the dashboard should of course support this functionality too, to avoid a "feature gap". But we need to resolve such requests in the correct order.

Reply:
I think that the problem now is more in the UI, we have the basic things needed in the orchestrator to create the OSDs. 

In any case, the important thing here is to agree with a design to create OSDs